### PR TITLE
Dependencies indicator type

### PR DESCRIPTION
**Pull request by teizenman**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
Related: https://github.com/demisto/etc/issues/24610

## Description
Added the fields to id_set

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/commands/find_dependencies/tests/find_dependencies_test.py [hunk 1]
```diff
@@ -377,6 +380,94 @@ def test_collect_detection_of_optional_dependencies_in_playbooks(self, integrati
             assert not found_result[1]  # validate that mandatory is set to False
 
 
+class TestDependsOnLayout:
```
Comment (ronykoz) [inline demisto_sdk/commands/find_dependencies/tests/find_dependencies_test.py:383]: let's use the when given then

File: demisto_sdk/commands/find_dependencies/find_dependencies.py [hunk 2]
```diff
@@ -291,6 +291,120 @@ def _collect_playbooks_dependencies(pack_playbooks, id_set):
...
+            id_set (dict): id set json.
+
+        Returns:
+            set: dependencies data that includes pack id and whether is mandatory or not.
+
+        """
+        dependencies_packs = set()
+
+        for layout in pack_layouts:
+            layout_data = next(iter(layout.values()))
```
Comment (ronykoz) [inline demisto_sdk/commands/find_dependencies/find_dependencies.py:310]: why do we need next and iter

File: demisto_sdk/commands/find_dependencies/find_dependencies.py [hunk 3]
```diff
@@ -291,6 +291,120 @@ def _collect_playbooks_dependencies(pack_playbooks, id_set):
...
+                    _label_as_mandatory(packs_found_from_incident_indicator_types)
+                dependencies_packs.update(pack_dependencies_data)
+
+            related_incident_and_indicator_fields = layout_data.get('incident_and_indicator_fields', [])
+            packs_found_from_incident_indicator_fields = PackDependencies._search_packs_by_items_names(
+                related_incident_and_indicator_fields, id_set['IncidentFields'] + id_set['IndicatorFields'])
+
+            if packs_found_from_incident_indicator_fields:
+                pack_dependencies_data = PackDependencies. \
+                    _label_as_mandatory(packs_found_from_incident_indicator_fields)
```
Comment (ronykoz) [inline demisto_sdk/commands/find_dependencies/find_dependencies.py:327]: are we sure it's mandatory?

File: demisto_sdk/commands/find_dependencies/find_dependencies.py [hunk 4]
```diff
@@ -291,6 +291,120 @@ def _collect_playbooks_dependencies(pack_playbooks, id_set):
...
+        Returns:
+            set: dependencies data that includes pack id and whether is mandatory or not.
+
+        """
+        dependencies_packs = set()
+
+        for incident_field in pack_incidents_fields:
+            incident_field_data = next(iter(incident_field.values()))
+
+            related_incident_types = incident_field_data.get('incident_types', [])
```
Comment (ronykoz) [inline demisto_sdk/commands/find_dependencies/find_dependencies.py:350]: incident field is pointing at type and vice versa?

File: demisto_sdk/commands/common/update_id_set.py [hunk 5]
```diff
@@ -302,6 +364,90 @@ def get_layout_data(path):
...
+    associated_types = json_data.get('associatedTypes')
+    if associated_types:
+        all_associated_types = set(associated_types)
+
+    system_associated_types = json_data.get('systemAssociatedTypes')
+    if system_associated_types:
+        all_associated_types = all_associated_types.union(set(system_associated_types))
+
+    if 'all' in all_associated_types:
+        all_associated_types = [list(incident_type.keys())[0] for incident_type in incidents_types_list]
```
Comment (ronykoz) [inline demisto_sdk/commands/common/update_id_set.py:395]: don't we just need to change that to ['all'] ?

File: demisto_sdk/commands/find_dependencies/find_dependencies.py (hunk 2 above)
Comment (teizenman) [reply demisto_sdk/commands/find_dependencies/find_dependencies.py]: Its just an easy way to get the data of the layout.
Thats how it is done in the rest of the code, so I wanted to keep the structure aligned

File: demisto_sdk/commands/common/update_id_set.py (hunk 5 above)
Comment (teizenman) [reply demisto_sdk/commands/common/update_id_set.py]: The idea is to create the id_set with the entire information so the dependency graph will only have to traverse over the data.

File: demisto_sdk/commands/find_dependencies/find_dependencies.py (hunk 4 above)
Comment (teizenman) [reply demisto_sdk/commands/find_dependencies/find_dependencies.py]: No, incidents fields are depended on incident types and thats it.

File: demisto_sdk/commands/find_dependencies/find_dependencies.py (hunk 3 above)
Comment (teizenman) [reply demisto_sdk/commands/find_dependencies/find_dependencies.py]: The file didnt say anything else

File: demisto_sdk/commands/common/update_id_set.py [hunk 6]
```diff
@@ -418,6 +418,43 @@ def get_incident_field_data(path, incidents_types_list):
...
 
+def get_indicator_type_data(path):
+    data = OrderedDict()
+    json_data = get_json(path)
+
+    id_ = json_data.get('id')
+    name = json_data.get('details', '')
+    fromversion = json_data.get('fromVersion')
+    toversion = json_data.get('toVersion')
+    associated_integration = json_data.get('reputationCommand')
```
Comment (ronykoz) [inline demisto_sdk/commands/common/update_id_set.py:429]: is it really an integration rather than a command?

Comment (teizenman) [reply demisto_sdk/commands/common/update_id_set.py:429]: Yes, its either `!ip/!email`,etc...

**Pull request merged**